types/@fortawesome/ember-fontawesome/fa-icon.d.ts

```diff
@@ -21,5 +21,7 @@ export interface FaIconComponentSignature {
     };
 }
 
-export interface FaIconComponent extends ArgsFor<FaIconComponentSignature> {}
-export declare class FaIconComponent extends Component<FaIconComponentSignature> {}
+declare class FaIconComponent extends Component<FaIconComponentSignature> {}
+
+export { FaIconComponent };
```

Comment:
You could do `export class FaIcon extends Component...` and drop the `export { FaIconComponent }`. Also, I don't think we need `declare` when we're in the d.ts.